src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -2638,6 +2593,39 @@ static void start_retriable_subchannel_batches(void* arg, grpc_error* ignored) {
 // LB pick
 //
 
+// The callback to intercept trailing metadata if retries is not enabled
+static void recv_trailing_metadata_ready_for_lb(void* arg, grpc_error* error) {
+  subchannel_batch_data* batch_data = static_cast<subchannel_batch_data*>(arg);
+  grpc_call_element* elem = batch_data->elem;
+  call_data* calld = static_cast<call_data*>(elem->call_data);
+  if (calld->pick.recv_trailing_metadata != nullptr) {
+    *calld->pick.recv_trailing_metadata =
+        batch_data->batch.payload->recv_trailing_metadata
+            .recv_trailing_metadata;
+  }
+  GRPC_CLOSURE_SCHED(
+      calld->pick.recv_trailing_metadata_ready,
+      GRPC_ERROR_REF(error));
+  GRPC_CLOSURE_RUN(
+      calld->original_recv_trailing_metadata_ready,
+      GRPC_ERROR_REF(error));
+}
+
+// Installs a interceptor to inform the lb of the trailing metadata, if needed
```

Comment:
Terminology nit: We don't refer to these callbacks as "interceptors"; that term refers to a higher-level interface in the C++ layer (or that of other wrapped languages).  In the context of this code, it's not unreasonable to use "intercept" as a generic verb, so I would say something like "If needed, intercepts the recv_trailing_metadata_ready callback to return trailing metadata to the LB policy".